tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/localedistance/LocaleDistanceMapper.java

```diff
@@ -434,21 +437,90 @@ private static RbValue ofMapEntries(Map<String, String> map) {
                 .elementsPerLine(2);
     }
 
-    // Returns an RbValue serialized from a sequence of LSR instance as a sequence of repeating
-    // (language, region, script) tuples (formatted as one tuple per line in the IcuData file).
-    //
-    // E.g.
-    // foo{
-    //     lang1, script1, region1,
-    //     ...
-    //     langN, scriptN, regionN,
-    // }
-    private static RbValue ofLsrs(Collection<LSR> lsrs) {
+    // Returns an RbValue serialized from a sequence of LSR instance as a sequence of number
+    // represent (language, region, script) tuples (formatted as one number per line in the IcuData file).
+    private static RbValue ofLsrNum(Collection<LSR> lsrs) {
         return RbValue.of(
                 lsrs.stream()
-                        .flatMap(lsr -> Stream.of(lsr.language, lsr.script, lsr.region))
-                        .collect(Collectors.toList()))
-                .elementsPerLine(3);
+                        .flatMapToInt(lsr -> IntStream.of(LSRToNum(lsr)))
+                        .mapToObj(Integer::toString));
+    }
+
+    // This method is added only to support encodeToIntForResource()
+    // It only support [a-z]{2,3} and will not work for other cases.
+    // TODO(ftang) Remove after LSR.encodeToIntForResource is available to the tool.
+    static private int encodeLanguageToInt(String language) {
+        assert language.length() >= 2;
+        assert language.length() <= 3;
+        assert language.charAt(0) >= 'a';
+        assert language.charAt(0) <= 'z';
+        assert language.charAt(1) >= 'a';
+        assert language.charAt(1) <= 'z';
+        assert language.length() == 2 || language.charAt(2) >= 'a';
+        assert language.length() == 2 || language.charAt(2) <= 'z';
+        return language.charAt(0) - 'a' + 1 +
+               27 * (language.charAt(1) - 'a' + 1) +
+               ((language.length() == 2) ? 0 : 27 * 27 * (language.charAt(2) - 'a' + 1));
+    }
+    // This method is added only to support encodeToIntForResource()
+    // It only support [A-Z][a-z]{3} which defined in UScript and does not work for other cases.
+    // TODO(ftang) Remove after LSR.encodeToIntForResource is available to the tool.
+    static private int encodeScriptToInt(String script) {
+        int ret = UScript.getCodeFromName(script);
+        assert ret != UScript.INVALID_CODE;
+        return ret;
+    }
+    // This method is added only to support encodeToIntForResource()
+    // It only support [A-Z]{2}|001|143|419 and does not work for other cases.
+    // TODO(ftang) Remove after LSR.encodeToIntForResource is available to the tool.
+    static private int encodeRegionToInt(String region) {
+        assert region.length() >= 2;
+        assert region.length() <= 3;
+        // Do not have enough bits to store the all 1000 possible combination of \d{3}
+        // Only support what is needed in resource- "001", "143" and "419".
```

Comment:
This is not the complete set. We also have for example 150 (in en_150). In RBNF data we used to have 003. Also I think it would be better to have at least an array of supported values, then just find the index of the correct element in that array (instead of the if-else sequence). And perhaps generate an error if you find a region code that is not in the list, because we could add a new numeric region at any time.

Reply:
well do. Changing the code now. Stay tune.

Reply:
DONE PTAL